packetbeat/sniffer/sniffer.go

```diff
@@ -51,7 +51,7 @@ type Sniffer struct {
 type sniffer struct {
 	config config.InterfaceConfig
 
-	state atomic.Int32 // store snifferState
+	state *atomic.Int32 // store snifferState
```

Comment:
Rather than making this a pointer, the child could be directly assigned to `sniffers[i]` at L105 and `child` be made a `*sniffer` from `sniffers[i]` for ergonomics immediately after. Your choice.